js-src/compat/plugins/woo-carrier-agents/checkout-woo-carrier-agents.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * Checkout scripts for: WooCommerce Carrier Agents (by Markup.fi).
+ */
+
+(function (root, factory) {
+	if ( typeof define === 'function' && define.amd ) {
+		define([], factory(root));
+	} else if ( typeof exports === 'object' ) {
+		module.exports = factory(root);
+	} else {
+		root.CheckoutWooCarrierAgents = factory(root);
+	}
+})(typeof global !== 'undefined' ? global : this.window || this.global, function (root) {
+
+	'use strict';
+
+	var $ = jQuery;
+	var _hasJQuery = ( $ != null );
+
+	var _hasInitialized = false;
+	var _publicMethods = {};
+	var _settings = {
+		searchFieldSelector: 'input[name="woo-carrier-agents-postcode"]',
+		enteredPostcode: '',
```

Comment:
The `_settings` object should only hold settings, not internal states or values. Instead, create a new variable `_enteredPostcode` directly to the main context:

```js
var _settings = {
	// ...
};
var _enteredPostcode = '';
```

Reply:
The value comes from PHP - I'm removing it from the settings declarations in JS.